spec/services/rdv_updater_spec.rb

```diff
@@ -0,0 +1,65 @@
+describe RdvUpdater, type: :service do
+  describe "#update" do
+    it "true when everything is ok" do
+      rdv = create(:rdv)
+      rdv_params = {}
+      expect(RdvUpdater.update(rdv, rdv_params)).to eq("Le rendez-vous a été modifié.")
+    end
+
+    it "call rdv.update with params" do
+      rdv = create(:rdv, context: "un contexte")
+      rdv_params = { context: "un autre context" }
+      expect(rdv).to receive(:update).with(rdv_params)
+      RdvUpdater.update(rdv, rdv_params)
+    end
+
+    it "return false when update fail" do
+      rdv = create(:rdv, agents: [create(:agent)])
+      rdv_params = attributes_for(:rdv, :future, agents: [])
+      expect(RdvUpdater.update(rdv, rdv_params)).to eq(false)
+    end
+
+    it "notify user when rdv cancelled" do
+      rdv = create(:rdv, status: "waiting")
+      rdv_params = { status: "excused" }
+      expect(Notifications::Rdv::RdvCancelledByAgentService).to receive(:perform_with).with(rdv)
+      RdvUpdater.update(rdv, rdv_params)
+    end
+
+    it "force updated_at to ensure new version will be recorded" do
+      now = Time.new(2020, 4, 23, 12, 56)
+      travel_to(now)
+      previous_date = now - 3.days
+
+      rdv = create(:rdv, updated_at: previous_date, context: "")
+      rdv_params = { context: "un nouveau context" }
+      RdvUpdater.update(rdv, rdv_params)
+      expect(rdv.reload.updated_at).to eq(now)
```

Comment:
je crois qu'il faut eviter les comparaisons strictes de dates, une fois stockées dans la db elles perdent un peu de precision, il vaut mieux utiliser `is_within(3.seconds).of(now)` syntaxe approximative